data/console/test_java.sh

```diff
@@ -0,0 +1,278 @@
+#!/bin/bash
+
+# Variables Definition
+# --------------------
+PWD=$(/usr/bin/pwd)
+DIR="/var/tmp"
+LIST_INSTALLED_IBM_VERSIONS="$DIR/javas_ibm"
+LIST_INSTALLED_JDK_VERSIONS="$DIR/javas_jdk"
+LIST_INSTALLED_GCJ_VERSIONS="$DIR/javas.gcj"
+RPM_QUERY_JAVA="$DIR/zypper_se_java"
+LIST_ALL_INSTALLED_VERSIONS="$DIR/javas_installed"
+HELLO_WORLD="$DIR/Hello.java"
+HELLO="$DIR/Hello"
+LIST_ALL_JAVA_ALTERNATIVES="$DIR/java_alternatives"
+LIST_ALL_JAVAC_ALTERNATIVES="$DIR/javac_alternatives"
+LIST_ALL_JAVAPLUGIN_ALTERNATIVES="$DIR/javaplugin_alternatives"
+
+# Function Definitions
+# --------------------
+# Test if there's any version of Java installed in the system
+is_java_installed() {
+    if [[ -s $LIST_ALL_INSTALLED_VERSIONS ]] ; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# Writes a very basic hello world program in Java language
+create_hello_java() {
+    # Construct the source code
+        cat << EOF > $HELLO_WORLD
+public class Hello
+{
+    public static void main (String argv[])
+    {
+        System.out.println(Hello World!);
```

Comment:
Double quotes are missing here, this won't compile, and will likely make your tests fail